### PR TITLE
Fix: Prevent NPE in SpecRunHistory.sortFeatures when duration is missing

### PR DESCRIPTION
**Pull request by ThanosTsiamis**
This PR fixes a NullPointerException in SpecRunHistory.sortFeatures that occurs when a .spock/RunHistory file contains a feature with a confidence value but no recorded duration (e.g., due to an interrupted or partially written test run). 

See Issue #2234 .

Previously, Spock assumed all featureDurations entries were present and performed direct unboxing to long, which fails when a value is null.

## Root cause
```
long duration1 = data.featureDurations.get(f1.getName());
long duration2 = data.featureDurations.get(f2.getName());
```

If either featureDurations.get(...) call returns null, the auto-unboxing causes:

`Cannot invoke "java.lang.Long.longValue()" because the return value of "java.util.Map.get(Object)" is null`

This typically happens when:

The previous run was aborted mid-write or during Gradle/IDE synchronization.

The .spock/RunHistory file was created by a different version of Spock.

The file was partially deleted or corrupted manually.

## Fix

A small defensive check ensures safe comparison even if a duration is missing:

```
Long d1 = data.featureDurations.get(f1.getName());
Long d2 = data.featureDurations.get(f2.getName());
if (d1 == null && d2 == null) return 0;
if (d1 == null) return 1;   // sort missing durations last
if (d2 == null) return -1;
return Long.compare(d1, d2);
```

This preserves existing behavior and avoids the NPE without changing sort semantics or serialization.

### CODE REVIEW

File: spock-core/src/main/java/org/spockframework/runtime/SpecRunHistory.java [hunk 1]
```diff
@@ -68,9 +68,13 @@ public void sortFeatures(SpecInfo spec) {
...
-      long duration1 = data.featureDurations.get(f1.getName()); // never null
-      long duration2 = data.featureDurations.get(f2.getName()); // never null
-      return duration1 < duration2 ? -1 : 1;
+      Long d1 = data.featureDurations.get(f1.getName());
+      Long d2 = data.featureDurations.get(f2.getName());
+
+      if (d1 == null && d2 == null) return 0;
+      if (d1 == null) return 1;
+      if (d2 == null) return -1;
+      return Long.compare(d1, d2);
```
Comment (AndreasTu) [inline spock-core/src/main/java/org/spockframework/runtime/SpecRunHistory.java]: Thank you very much for the contribution.

Could you please add a line in
`docs\release_notes.adoc`
in the section 
```
== 2.4 (tbd)
=== Misc

* Fix: Prevent NPE in SpecRunHistory.sortFeatures when duration is missing spockIssue:2234[]
```

Comment (AndreasTu) [inline spock-core/src/main/java/org/spockframework/runtime/SpecRunHistory.java]: The method did never return 0 (e.g. equal) for same values in the past, so we should not do that now.

```suggestion
      if (d1 == null) return 1;
      if (d2 == null) return -1;
      return d1< d2 ? -1 : 1;
```

Comment (ThanosTsiamis) [reply spock-core/src/main/java/org/spockframework/runtime/SpecRunHistory.java]: Added as indicated =]

**Review by AndreasTu — APPROVED** [review]
Thank you for the contribution.

**Comment by AndreasTu** [comment]
Note: Normally I would also request a test case for such an issue, but I acknowledge that this hard to test, due to the null checks of the first map above. So I would say the regression in code coverage here it justifyable.
